### PR TITLE
feat: Added custom hex colors for testing workflow

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Added workflow to check if there is any custom colors or hex codes is being used. This workflow will be checking for the anomalies in the entire file that has been changed by the developer.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/27839

#### Type of change
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: .github/workflows/client-build.yml [hunk 1]
```diff
@@ -106,6 +111,46 @@ jobs:
...
+        id: client_files
+        uses: umani/changed-files@v4.0.0
+        with:
+          repo-token: ${{ secrets.APPSMITH_CI_TEST_PAT }}
+          pattern: 'app/client/src/.*'
+          pr-number: ${{ inputs.pr }}
+
+      # Check all the newly added files are in ts  
+      - name: ADS compliant check
+        if: inputs.ads-compliant-check == 'true' && inputs.pr != 0 && github.pull_request.base.ref == 'release'
```
Comment (KelvinOm) [inline .github/workflows/client-build.yml:126]: Let's consider creating a custom rule for ESLint. This way we will be able to highlight errors in the development process or on a pre-commit hook. If we check this only in CI, then the development cycle time will be increased. This is a poor DX.

Comment (albinAppsmith) [reply .github/workflows/client-build.yml:126]: We need this to be a blocking change so that the inconsistencies will get resolved soon.

Comment (KelvinOm) [reply .github/workflows/client-build.yml:126]: Eslint checks are mandatory checks in CI as part of `Quality checks / client-lint` process.

**Comment by KelvinOm** [comment]
> 🔴 Below files are not compliant with ADS. Please fix and re-trigger ok-to-test
> 
> 1. app/client/src/App.css
> 2. app/client/src/index.css
> 3. app/client/src/pages/Applications/ApplicationCard.tsx

@nidhi-nair What do you think about making the same message for Quality checks?

**Review by sarojsarab — APPROVED** [review]

**Pull request merged**